pr_agent/algo/utils.py

```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import json
+import logging
+import re
 import textwrap
 
```

Comment:
The function `try_fix_json` is doing more than one thing: it tries to fix the JSON and also logs an error if it can't. Consider separating these responsibilities into different functions to adhere to the Single Responsibility Principle. [important]